boa/src/exec/tests.rs

```diff
@@ -1235,3 +1235,135 @@ fn not_a_function() {
     "#;
     assert_eq!(forward(&mut engine, scenario), "TypeError: not a function");
 }
+
+#[test]
+fn assignment_to_non_assignable() {
+    // Relates to the behaviour described at
+    // https://tc39.es/ecma262/#sec-assignment-operators-static-semantics-early-errors
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+
+    const ERR_MSG: &str =
+        "Parsing Error: Syntax Error: Syntax Error: Invalid left-hand side in assignment";
```

Comment:
Why is it `"Parsing Error: Syntax Error: Syntax Error: Invalid left-hand side in assignment"`?

shouldn't a `SyntaxError` be thrown with `.throw_syntax_error()` when there is a syntax error. #293 should make it easier to test for syntax error since it prints it `SyntaxError: message`, not all the fields for the error object.

Reply:
This error originates in the parser/lexer though so it is before the interpreter (and therefore .throw_syntax_error)?

Reply:
In `forward` we parse and we also have the Interpreter as an parameter so we can call `.throw_syntax_error()`. but this is kind of pointless since return a `String`.

Why does `forward` return a string it does not make sense to me, it should `forwrad` the value (`ResultValue`) not convert it to a string, `forward_val` does that, which does not make sense either.
The default behaviour should be to forward the value and if we need a string we will convert it.

^ this is kind of unrelated to this issue, but could we at least match the out push string `SyntaxError: message`, and not `"Parsing Error: Syntax Error: Syntax Error: Invalid left-hand side in assignment"`

Reply:
Ok yeah I understand what you mean, I'll switch the hardcoded expected message to instead just look for the message to contain 'SyntaxError: '.